Kernel, u-boot and omxplayer improvements

betafive opened the pull request:
* Fix omxplayer build so that we don't need the internet during do_compile.

* Refactor the kernel recipes, merging the two inc files together in the process. Drop duplicate, outdated and incorrect code from the recipes.

* Update linux-raspberrypi-dev to the 4.13.y series.

* Support u-boot on raspberrypi3-64. This requires a change to the method of enabling u-boot as we can no longer just infer this from the KERNEL_IMAGETYPE value. Therefore a new RPI_USE_U_BOOT variable is used.

* Fix some minor issues in sdcard_image-rpi.bbclass found whilst implementing the above.

betafive commented:
Boot tested on raspberrypi3 (32-bit) and raspberrypi3-64, with and without u-boot in both cases. Everything looks good. On 32-bit I've even confirmed that we can use zImage with u-boot instead of uImage if we want to.

If this is too big to review in one go let me know and I'll split it up further. It's been a productive Friday :)

betafive commented:
Updated after testing with wic images. It turns out there was some dependency information missing.

agherzan commented on recipes-multimedia/omxplayer/omxplayer/0006-Prevent-ffmpeg-configure-compile-race-condition.patch:
```diff
@@ -0,0 +1,39 @@
+the configure stage is finished before the compile stage starts.
+
+Signed-off-by: Paul Barker <pbarker@toganlabs.com>
+Upstream-status: Inappropriate
```
This doesn't sound like an inappropriate patch. Would it make sense to upstream it?

betafive replied:
I was a bit unsure on what to put as Upstream-status here. We should try to upstream a fix for this and I've opened the bug https://github.com/popcornmix/omxplayer/issues/567 to work towards that. However this patch is based on the previous one and can't be upstreamed exactly as-is.

agherzan replied:
This sounds like Pending, doesn't it?

betafive replied:
Ok cool I'll change it to Pending.

betafive commented:
I've updated the Upstream-status tag and added a comment to explain the use of ffmpeg in the omxplayer recipe as I realised it wasn't immediately obvious.

agherzan commented:
@betafive could you please rebase this? Looks good.

betafive commented:
Rebase done, I'm rebuilding this and plan to re-do a couple of boot tests.

betafive commented:
Rebase done, all now up to date. I'm still having issues with HMDI output on raspberrypi3-64 but that's not related to these changes.

agherzan commented:
Thanks @betafive . Merged to master.
